File: java/lib/src/main/java/algorithm/sorting/SelectionSort.java
package algorithm.sorting;

public class SelectionSort extends Sort {
  @Override
  protected int[] sort(int[] arr, SortType type) {
    int length = arr.length;

    for (int i = 0; i < length - 1; i++) {
      int minIndex = i;
      for (int j = i + 1; j < length; j++) {
        if (super.compare(arr[minIndex], arr[j], type)) {
          minIndex = j;
        }
      }
      super.swap(arr, minIndex, i);
    }
    return arr;
  }
}

File: kotlin/lib/src/main/kotlin/algorithm/sorting/SelectionSort.kt
package algorithm.sorting

public class SelectionSort : Sort() {
  override fun sort(arr: Array<Int>, type: SortType): Array<Int> {
    val length = arr.size

    for (i in 0..length - 1) {
      var minIndex = i

      for (j in i + 1..length - 1) {
        if (compare(arr[minIndex], arr[j], type)) {
          minIndex = j
        }
      }
      arr.swap(minIndex, i)
    }
    return arr
  }
}
